schmatic.md: UART_TX and UART_RX pins are falsy labled

jkoeppeler opened the pull request:
According to this commit message ( https://github.com/nrf-rs/microbit/commit/10484c6fa485c259d3a67ae62312d24d0f94874c )  and testing this myself, the pins are falsely labeled. The RX pin should be TX and vice versa.

sagrawal-mips commented:
@microbit-carlos Can you please merge this merge request? This will help us new folks a lot.

microbit-carlos commented:
These are the docs for the schematic, and so they follow the naming convention used there, which in this case present the internal UART from the perspective of the Interface chip.

<img width="284" alt="image" src="https://github.com/user-attachments/assets/9ead6b76-714e-49de-8ec9-2ddfb8a88a4e" />
<br><br>
<img width="255" alt="image" src="https://github.com/user-attachments/assets/36487cce-47c1-4169-8e33-63b59b014137" />

I know this is unfortunate, as the majority of developers will look at these signals from the perspective of the Target MCU.
We've taken this as a lesson learned, and in the future we'll use longer and more clear names, like `UART_INT_TARGET_TX_INTERFACE_RX`, but for the time being is better to keep the references in sync between the schematic and the docs.

That being said, we should definitely add a note!
@jkoeppeler or @sagrawal-mips would you be happy to add a `*` next to the pin names and a clarification below the table?

jkoeppeler commented:
@microbit-carlos @sagrawal-mips I updated the description and added the footnote. Let me know if you like it, or if I should change it.
And thanks a lot for the clarification on that issue.

microbit-carlos commented on hardware/schematic.md:
```diff
@@ -75,6 +75,8 @@ Below is the pinmap and allocation of the nRF52833, more information is availabl
 | P0.31            | COL3                | N                        | P3                  |
 | P0.30            | COL5                | N                        | P10                 |
 
+[^*]: These names follow the schematic's naming convention, labeled from the interface's perspective, not the target MCU's. So, UART_INT_TX corresponds to the MCU's RX pin, and UART_INT_RX to the MCU's TX pin.
```
@microbit-robert do you know if the current Markdown parser for the Tech Site Jekyll builds (is it still using Jekyll?) accepts the footnotes notation?

jkoeppeler replied:
yeah I just tried it, and it places the footnote at the end of the page, which I don't like. I will change to the `*` as you suggested as its the better way I guess.

microbit-carlos replied:
> yeah I just tried it, and it places the footnote at the end of the page, which I don't like. I will change to the * as you suggested as its the better way I guess.

Ah, good point, thanks for checking, I agree it's better with the note below the table!

jkoeppeler replied:
Just let me know if there's anything left or requires changes. Happy to apply them :)
